File: Segundo/Monopoly/civitas_java (1)/proyecto_civitas/src/civitas/SorpresaIrCasilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;
/**
 *
 * @author pc
 */
public class SorpresaIrCasilla extends Sorpresa{
    private int valor;
    private Tablero tablero;
     
    SorpresaIrCasilla(Tablero tablero,int valor,String texto)
    {
        super(texto);
        this.valor=valor;
        this.tablero=tablero;
    }
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos)
    {
        if(jugadorCorrecto(actual,todos))
        {
            informe(actual,todos);
            int a=todos.get(actual).getNumCasillaActual();
            int tirada=tablero.calcularTirada(a,valor);
            int nuevaPos=tablero.nuevaPosicion(a,tirada);
            todos.get(actual).moverACasilla(nuevaPos);
            tablero.getCasilla(nuevaPos).recibeJugador(actual, todos);    
        }
        
    }
    @Override
    public String toString() {
        return "SorpresaIrCasilla: " + "valor: " + valor + ", tablero: " + tablero +super.toString() +'\n';
    }

}

File: Segundo/Monopoly/civitas_ruby (1)/proyecto_civitas_ruby/lib/sorpresa_ir_casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Civitas
class SorpresaIrCasilla < Sorpresa
  def initialize(texto,tablero,valor)
    super(texto)
    @tablero = tablero
    @valor = valor
  end
  
  def aplicarAJugador(actual,todos)
      if(jugadorCorrecto(actual,todos))
        super(actual,todos)
        casilla_actual=todos[actual].getNumCasillaActual
        tirada=@tablero.calcularTirada(casilla_actual,@valor)
        nueva_pos=@tablero.nuevaPosicion(casilla_actual,tirada)
        todos[actual].moverACasilla(nueva_pos)
        @tablero.getCasilla(nueva_pos).recibeJugador(actual, todos)
      end
  end
  
  def toString
    msg = "Ir Casilla"
    return msg
  end
  public_class_method :new
end
end